src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLLayer.m

```diff
@@ -122,7 +123,40 @@ - (void) blitTexture {
                 sourceSize:MTLSizeMake(src_w, src_h, 1)
                 toTexture:mtlDrawable.texture destinationSlice:0 destinationLevel:0 destinationOrigin:MTLOriginMake(0, 0, 0)];
         [blitEncoder endEncoding];
+#else
+        id<MTLCommandBuffer> cb = [ctx createCommandBuffer];
+        id<MTLComputeCommandEncoder> computeEncoder = [cb computeCommandEncoder];
+        id<MTLComputePipelineState> computePipelineState = [ctx.pipelineStateStorage
+                                                            getComputePipelineState:@"tex2tex_opaque"];
+        [computeEncoder setComputePipelineState:computePipelineState];
+
+        NSUInteger maxTotalThreadsPerThreadgroup = computePipelineState.maxTotalThreadsPerThreadgroup;
+        NSUInteger w = computePipelineState.threadExecutionWidth;
+
+        // Workaround for some OS/device bug reporting incorrect maxTotalThreadsPerThreadgroup
+        if (maxTotalThreadsPerThreadgroup == 0) {
+            maxTotalThreadsPerThreadgroup = 1;
+            w = 1;
+        }
+
+        NSUInteger h = maxTotalThreadsPerThreadgroup / w;
+        MTLSize threadgroupSize = MTLSizeMake(w, h, 1);
+        MTLSize threadgroupCount;
+
+        threadgroupCount.width  = (buffer.width - src_x + threadgroupSize.width - 1) / threadgroupSize.width;
+        threadgroupCount.height = (buffer.height - src_y + threadgroupSize.height - 1) / threadgroupSize.height;
+        threadgroupCount.depth = 1;
+
+        [computeEncoder setTexture:buffer atIndex:0];
+        [computeEncoder setTexture:mtlDrawable.texture atIndex:1];
+        struct InsetsUniforms uniforms = {src_x, src_y};
+        [computeEncoder setBytes:&uniforms length:sizeof(struct InsetsUniforms) atIndex:2];
 
+        [computeEncoder dispatchThreadgroups:threadgroupCount
+                               threadsPerThreadgroup:threadgroupSize];
+        [computeEncoder endEncoding];
+        [cb commit];
+#endif
```

Comment:
This is better than changing the background color, but should be carefully checked. @jayathirthrao @aghaisas please take a look.
Since this affects every blit to the window I suggest running a full test run.

Reply:
@jayathirthrao @aghaisas please let me know if I can push this change into JDK17 repository

Reply:
@avu i have given all test run. I will get back to you once it is done.

Reply:
Automated jtreg/JCK test run is green.

Reply:
@avu If all our automated tests are green but we still have some issues, then an additional automated test needs to be added.